spec/factories.rb

```diff
@@ -20,8 +21,11 @@
     end
   end
 
+  factory :step_by_step_page_with_steps_but_no_draft, parent: :step_by_step_page_with_steps do
+    draft_updated_at { nil }
```

Comment:
Is this possible? Unless the step-by-step has been unpublished, there should always be a `draft_updated_at` date.

Reply:
What do you mean by "no draft"? Does that mean the step by step is published?

Reply:
I did this because of [this test](https://github.com/alphagov/collections-publisher/blob/master/spec/models/step_by_step_page_spec.rb#L345-L347):

```ruby
    it 'cannot be published if it does not have a draft' do
      expect(step_by_step_page.can_be_published?).to be false
    end
```

Now that the default factory sets the `draft_updated_at` field, this was returning `true`. But what I could do is call `mark_as_published` in this step before calling `can_be_published?`?

Reply:
This test expects `has_draft?` to return false.

Reply:
`mark_as_published` would mean there is no draft, therefore `can_be_published?` becomes `false`.

Can you explain what your proposal is?

Reply:
Is it that we should be using `published_step_by_step_page` instead? (Agreed that makes more sense as a step by step with steps but no draft is not possible).

Reply:
This test is checking the first part of this condition: https://github.com/alphagov/collections-publisher/blob/cd98e0e72ff212144d04242b487f58d61d906c37/app/models/step_by_step_page.rb#L94 that `has_draft?` is false. 

So, yes using `published_step_by_step_page` should work, as long as the conditions that `has_draft?` relies on haven't changed.